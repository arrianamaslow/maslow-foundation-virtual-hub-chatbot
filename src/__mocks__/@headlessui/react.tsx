/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
