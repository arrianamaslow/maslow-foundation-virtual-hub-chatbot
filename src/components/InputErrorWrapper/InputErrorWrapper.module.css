.wrapper {
  @apply relative;
}

.redBorder {
  @apply [&>*]:!border-[#FF0000];
}

.errorMessageText {
  @apply text-[#FF0000] text-[1rem];
}
