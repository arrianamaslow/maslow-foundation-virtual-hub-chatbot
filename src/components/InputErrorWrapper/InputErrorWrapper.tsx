import React, { Suspense } from 'react'

import style from './InputErrorWrapper.module.css'
import Image from 'next/image'

export type InputErrorWrapperProps = {
  children: string | JSX.Element | JSX.Element[]
  errorMessage?: string
}

const InputErrorWrapper: React.FC<InputErrorWrapperProps> = ({ errorMessage, children }) => {
  return (
    <Suspense>
      <div data-testid='outerBox' className={`${style.wrapper} ${errorMessage && style.redBorder}`}>
        {children}
        {errorMessage && (
          <div className='absolute top-0 right-0 h-full flex items-center justify-center px-[16px]'>
            <div>
              <Image
                width={32}
                height={32}
                alt='Exclamation mark'
                src='/img/exclamation-mark.svg'
              />
            </div>
          </div>
        )}
      </div>
      <div className={style.errorMessageText}>{errorMessage}</div>
    </Suspense>
  )
}

export default InputErrorWrapper
