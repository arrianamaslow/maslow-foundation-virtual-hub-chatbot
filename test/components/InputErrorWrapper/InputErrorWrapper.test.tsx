/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react'
import InputErrorWrapper, {
  InputErrorWrapperProps
} from '@/components/InputErrorWrapper/InputErrorWrapper'
import '@testing-library/jest-dom'
jest.mock('@/components/CheckBox/CheckBox', () => ({ tickIsActive }: { tickIsActive: boolean }) => {
  ;<div>{tickIsActive ? 'âœ”' : ''}</div>
})

const DEFAULT_ERROR_MESSAGE = 'ERROR'

const defaultProps: InputErrorWrapperProps = {
  children: <p>CHILDREN</p>,
  errorMessage: DEFAULT_ERROR_MESSAGE
}

describe('<InputErrorWrapper />', () => {
  it('should render as expected', () => {
    const { container } = render(<InputErrorWrapper {...defaultProps} />)

    expect(container).toMatchSnapshot()
  })

  it('displays error message and image', () => {
    const { container } = render(<InputErrorWrapper {...defaultProps} />)
    const errorImage = screen.getByAltText('Exclamation mark')

    expect(container.textContent).toContain(DEFAULT_ERROR_MESSAGE)
    expect(errorImage).toHaveAttribute('src', '/img/exclamation-mark.svg')
  })

  it('does not display an error message or image when error message prop is empty', () => {
    const { container } = render(<InputErrorWrapper {...defaultProps} errorMessage='' />)

    expect(container.textContent).not.toContain(DEFAULT_ERROR_MESSAGE)
    expect(container.getElementsByTagName('img').length).toEqual(0)
  })

  it('displays a red border around the input when there is an error message', () => {
    render(<InputErrorWrapper {...defaultProps} />)

    const outerBoxElement = screen.getByTestId('outerBox')
    expect(outerBoxElement).toHaveClass('redBorder')
  })

  it('does not display a red border around the input when there is no error message', () => {
    render(<InputErrorWrapper {...defaultProps} errorMessage='' />)

    const outerBoxElement = screen.getByTestId('outerBox')
    expect(outerBoxElement).not.toHaveClass('redBorder')
  })
})
