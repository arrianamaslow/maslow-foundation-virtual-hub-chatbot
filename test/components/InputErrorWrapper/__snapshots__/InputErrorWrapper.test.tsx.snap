// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputErrorWrapper /> should render as expected 1`] = `
<div>
  <div
    class="wrapper redBorder"
    data-testid="outerBox"
  >
    <p>
      CHILDREN
    </p>
    <div
      class="absolute top-0 right-0 h-full flex items-center justify-center px-[16px]"
    >
      <div>
        <img
          alt="Exclamation mark"
          height="32"
          src="/img/exclamation-mark.svg"
          width="32"
        />
      </div>
    </div>
  </div>
  <div
    class="errorMessageText"
  >
    ERROR
  </div>
</div>
`;
